addons/nodot/interaction/Teleport3D.gd

```diff
@@ -0,0 +1,82 @@
+##  Teleport3D is an Area3D which, when touched,
+##  teleports the collider to another location either 
+##  instantly or after a defined time
+class_name Teleport3D extends Area3D
+
+@export_category("Nodes")
+##  Target node for copy the coords
+@export_node_path("Node3D") var target
+##  Reference a node to teleport by target
+@export_node_path("CharacterBody3D") var player
+
+@export_category("Others Variables")
+##  Size of collision area
+@export var size = Vector3(3.372, 3.821, 1)
+##  Time to teleport (in seconds, and this variable is optional)
+@export var time_teleport : int = 0
+
+##  Get position of "Player" 
+@onready var node_player = get_node(player)
+##  Get position of "Target"
+@onready var position_target = get_node(target)
+
+##  Collision Shape
+var node_collision = CollisionShape3D.new()
+##  Timer node
+var node_time = Timer.new()
+
+##  Check variable if "Player" is in "area"
+var can_teleport_area = false
+##  Check variable, if "Player" wait a time
+var can_teleport_time = false
+
+func _enter_tree():
+	##  add collision to scene
```

Comment:
Only root level comments need the double hash. (It's just for the docs generator)